### PR TITLE
feat(eslint): ESLint now makes a best effort to sort the extends array

### PR DESCRIPTION
**Pull request by dkershner6**
Fixes https://github.com/projen/projen/issues/2806

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/javascript/eslint.test.ts [hunk 1]
```diff
@@ -104,18 +104,46 @@ test("if the prettier is configured, eslint is configured accordingly", () => {
...
+  });
+
+  eslint.addExtends("plugin:some-plugin/recommended");
+
+  // THEN
+  const output = synthSnapshot(project);
+
+  const extendsArray = output[".eslintrc.json"].extends;
+  expect(extendsArray.at(-1)).toEqual("plugin:some-plugin/recommended");
+  expect(extendsArray).toContain("plugin:prettier/recommended");
```
Comment (mrgrain) [inline test/javascript/eslint.test.ts:144]: This test case is not super obvious. Instead, let's match the actual value here. Should not be too long are change to often.

File: src/javascript/eslint.ts [hunk 2]
```diff
@@ -62,6 +61,15 @@ export interface EslintOptions {
...
   readonly prettier?: boolean;
 
+  /**
+   * The extends array in eslint is order dependent.
+   * Setting this option true will make a best effort to properly sort the extends array
+   * according to ESLint best practices.
+   *
+   * @default true
+   */
+  readonly sortExtends?: boolean;
```
Comment (mrgrain) [inline src/javascript/eslint.ts:71]: I wonder if we should just allow a compare function here:
```suggestion
  readonly sortExtends?: ICompareString;
```

```ts
interface ICompareString {
  compare(a: string, b: string): number;
}
```

And our Eslint component can provide a default:

class Eslint {
  static public SORT_EXTENDS_DEFAULT(): ICompareString {
    return {
      compare(a: string, b: string): number {
          /// to do
      }
    };
  }
}

Comment (dkershner6) [reply src/javascript/eslint.ts:71]: I like this idea.  I assume the nesting is a JSII requirement, as I see the CDK do that too, and I can't imagine .NET or java having a function parameter...

Comment (mrgrain) [reply src/javascript/eslint.ts:71]: Yes, unfortunately free-floating functions are not jsii compatible. So we need a protocol interface for it.

**Comment by dkershner6** [comment]
OK, @mrgrain , we now have a sorting function and an extra test.

File: src/index.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export * from "./task-runtime";
 export * from "./testing";
 export * from "./textfile";
 export * from "./toml";
+export * from "./util/types";
```
Comment (mrgrain) [inline src/index.ts:36]: Let's go with
```suggestion
export * from "./compare;
```

File: src/javascript/eslint.ts [hunk 4]
```diff
@@ -63,12 +64,11 @@ export interface EslintOptions {
 
   /**
    * The extends array in eslint is order dependent.
-   * Setting this option true will make a best effort to properly sort the extends array
-   * according to ESLint best practices.
+   * This option allows to sort the extends array in any way seen fit.
    *
-   * @default true
+   * @default Use known ESLint best practices to place "prettier" plugins at the end of the array
```
Comment (mrgrain) [inline src/javascript/eslint.ts:69]:
```suggestion
   * @default - Use known ESLint best practices to place "prettier" plugins at the end of the array
```

**Comment by dkershner6** [comment]
Updated, @mrgrain .

**Comment by dkershner6** [comment]
Another @mrgrain nudge.

**Comment by mrgrain** [comment]
Still on my list. Should get to it this week. Sorry for the delay.

**Pull request merged**